### ISSUE TITLE
ConcurrentModificationException occurred with sample()

### ISSUE THREAD

**Issue by mhyeon-lee** [jlink/jqwik#205]
## Testing Problem

The code below will cause ConcurrentModificationException.

```java
Arbitraries.strings()
                .map(it -> it + Arbitraries.strings().alpha().sample())
                .filter(StringUtils::isNotBlank)
                .sample()
```

<img width="1140" alt="스크린샷 2021-07-06 오전 12 10 40" src="https://user-images.githubusercontent.com/5206572/124491779-9bc0f080-ddee-11eb-850c-a3e2299145bc.png">

Occurs when modifying static generators from SampleStreamFacade to recursive.

## Discussion

Error does not occur when declaring withoutEdgeCases as below code.

```java
Arbitraries.strings()
                .map(it -> it + Arbitraries.strings().alpha().sample())
                .filter(StringUtils::isNotBlank)
                .withoutEdgeCases()
                .sample()
```

I'm using it temporarily as above, but could you fix this problem?

**Comment by jlink** [comment]
Thanks for the catch. At first glance it looks like a bug. I’ll examine the problem.

**Comment by jlink** [comment]
That being said, your code is most probably not what you should do anyway.
If you want to concatenate two random strings the most obvious way is:

```
Combinators.combine(Arbitraries.strings(), Arbitraries.strings().alpha())
    .as((s1, s2) -> s1 + s2)
    .filter(…)
```
The rule is to never use sample() within a generator; it will not end well  ;-)

**Comment by jlink** [comment]
@mhyeon-lee I cannot reproduce the problem in version 1.5.2. Which version are you using?

Maybe there's more going on than what can be seen in your code example?

**Comment by mhyeon-lee** [comment]
@jlink 
Here is my fail test case.

https://github.com/mhyeon-lee/jqwik/pull/1/files#diff-fecbf655309e9b4842780db53963942742decb72ae65f71f4698ec1814ba4d3cR10

**Comment by jlink** [comment]
It's got to do with the Java version. Java 8, which I use locally since it's the minimum version, runs fine. Java >= 11 shows the error.

**Comment by jlink** [comment]
Hopefully fixed in https://github.com/jlink/jqwik/commit/fcdcfcad65df84cc8c087093ddc14be489bd5e1b

**Comment by jlink** [comment]
@mhyeon-lee Thanks again for catching the bug. But don't do it that way!

Available in 1.5.3-SNAPSHOT

**Comment by jlink** [comment]
Released in 1.5.3
